+nosnoc/+discrete_problem/stewart.m

```diff
@@ -0,0 +1,771 @@
+            obj.w.lambda(0,0,opts.n_s) = {{['lambda'], dims.n_lambda},0,inf};
+            obj.w.lambda(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp)) = {{'lambda', dims.n_lambda},0, inf};
+            obj.w.theta(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s)) = {{'theta', dims.n_theta},0, 1};
+            obj.w.mu(0,0,opts.n_s) = {{'mu', dims.n_mu},0,inf};
+            obj.w.mu(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp)) = {{'mu', dims.n_mu},0,inf};
+
+            % Handle x_box settings
+            if ~opts.x_box_at_stg && opts.irk_representation ~= IrkRepresentation.differential
+                obj.w.x(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp-1)).lb = -inf*ones(dims.n_x, 1);
+                obj.w.x(1:opts.N_stages,1:opts.N_finite_elements(1),1:(opts.n_s+rbp-1)).ub = inf*ones(dims.n_x, 1);
+            end
+
+            if ~opts.x_box_at_fe
+                obj.w.x(1:opts.N_stages,1:(opts.N_finite_elements(1)-1),opts.n_s+rbp).lb = -inf*ones(dims.n_x, 1);
+                obj.w.x(1:opts.N_stages,1:(opts.N_finite_elements(1)-1),opts.n_s+rbp).ub = inf*ones(dims.n_x, 1);
+            end
+        end
+
+        function forward_sim_constraints(obj)
+            import casadi.*
+            model = obj.model;
+            opts = obj.opts;
+            dcs = obj.dcs;
+            dims = obj.dcs.dims;
+
+            rbp = ~opts.right_boundary_point_explicit;
+
+            if obj.opts.use_fesd
+                t_stage = obj.p.T()/opts.N_stages;
+                h0 = obj.p.T().val/(opts.N_stages*opts.N_finite_elements(1));
+            else
+                h0 = obj.p.T().val/(opts.N_stages*opts.N_finite_elements(1));
+            end
+
+            v_global = obj.w.v_global();
+            p_global = obj.p.p_global();
+
+            x_0 = obj.w.x(0,0,opts.n_s);
+            z_0 = obj.w.z(0,0,opts.n_s);
+            lambda_0 = obj.w.lambda(0,0,opts.n_s);
+            mu_0 = obj.w.mu(0,0,opts.n_s);
+
+            obj.g.z(0,0,opts.n_s) = {dcs.g_z_fun(x_0, z_0, obj.w.u(1), v_global, [p_global;obj.p.p_time_var(1)])};
+            obj.g.switching(0,0,opts.n_s) = {dcs.g_switching_fun(x_0, z_0, lambda_0, mu_0, v_global, [p_global;obj.p.p_time_var(1)])};
+            
+
+            x_prev = obj.w.x(0,0,opts.n_s);
+            for ii=1:opts.N_stages
+                ui = obj.w.u(ii);
+                p_stage = obj.p.p_time_var(ii);
+                p = [p_global;p_stage];
+                if obj.opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
+                    s_sot = obj.w.sot(ii);
+                elseif obj.opts.use_speed_of_time_variables
+                    s_sot = obj.w.sot();
+                else
+                    s_sot = 1;
+                end
+
+                sum_h = 0;
+                for jj=1:opts.N_finite_elements(ii)
```

Comment:
here N_finite_elemnts  as vector seems to be supported - maybe we addapt it above as well